privacy_estimates/experiments/attacks/dpd/data.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass, asdict
+from typing import List
+from pathlib import Path
+from json import load, dump
+from opacus.optimizers import DPOptimizer
+
+from .attack import CanaryTrackingOptimizer
+
+
+@dataclass
+class DPDistinguishingData:
+    scores: List[float]
+    sensitivity: float
+
+    @classmethod
+    def from_opacus(cls, optimizer: DPOptimizer) -> "DPDistinguishingData":
```

Comment:
When you add type annotations and also check the type, maybe it is time to consider a statically typed language 😄